### PR TITLE
[top, fpga] Rework SCA/FI trigger signal selection and generation

### PR DESCRIPTION
**Pull request by vogelpi**
This PR modifies the SCA/FI trigger signal generation on FPGA to also add the option for a fully software-controlled trigger. This is required to prepare the SCA capture setup for the penetration testing of the ES silicon.

This is related to lowRISC/ot-sca#198.

I've successfully tested the CW310 bitstream and the AES binary generated from this PR. What's left to change / align is the CW305 as well as KMAC and OTBN binaries.

### CODE REVIEW

File: sw/device/sca/lib/sca.c [hunk 1]
```diff
@@ -109,7 +115,7 @@ static void sca_init_gpio(sca_trigger_source_t trigger) {
 
   uint32_t select_mask =
       bitfield_field32_write(0, kTriggerSourceBitfield, UINT32_MAX);
-  uint32_t enable_mask = bitfield_bit32_write(0, kTriggerGateBitIndex, true);
+  uint32_t enable_mask = bitfield_bit32_write(0, kTriggerHwGateBitIndex, true);
```
Comment (vogelpi) [inline sw/device/sca/lib/sca.c:118]: Right now, we always use the precise hardware-generated trigger. I think in the short term we should add simple_serial command to change this at runtime. I will work on this on THU. Ah, and I also need to test the CW305 and update kmac/otbn_serial etc.

File: sw/device/sca/lib/sca.c [hunk 2]
```diff
@@ -44,11 +44,17 @@ static const bitfield_field32_t kTriggerSourceBitfield = {
 
 enum {
   /**
-   * Bit index of the trigger gate signal for gating the trigger from software.
+   * Bit index of the hardware trigger gate signal for gating the hardware trigger from software.
```
Comment (nasahlpa) [inline sw/device/sca/lib/sca.c:47]: It seems that linting failed as this line is too long.

**Review by nasahlpa — APPROVED** [review]
Thanks @vogelpi , looks good for me.

**Comment by vogelpi** [comment]
Update: I've now also implemented the command to switch the trigger, and tested all modes on the FPGAs. Related to that, I've also fixed a bug in AES fvsr batch mode occuring with latest bitstreams.

**Comment by msfschaffner** [comment]
Do I understand correctly that this won't impact any of the ASIC RTL? If yes, this is ok from my end.

**Review by msfschaffner — APPROVED** [review]

**Comment by msfschaffner** [comment]
One request: can you make sure this is compatible with https://github.com/lowRISC/opentitan/pull/20210?

**Comment by vogelpi** [comment]
> Do I understand correctly that this won't impact any of the ASIC RTL? If yes, this is ok from my end.

Yes, it doesn't impact ASIC RTL, just the FPGA chip-level.

**Comment by vogelpi** [comment]
> One request: can you make sure this is compatible with #20210?

Yes, just tried to locally rebase #20210 on top of this one, it works cleanly without invoking topgen again or similar. I am thus merging this.

**Pull request merged**